app/workers/sphinx_account_indexation_worker.rb

```diff
@@ -18,4 +18,16 @@ def perform(_model, id)
       delete_from_index(Account, id, *buyers)
     end
   end
+
+  protected
+
+  def reindex(instance)
+    ThinkingSphinx::Processor.new(instance: instance).upsert
+  end
+
+  def delete_from_index(model, *ids)
+    ids.each do |id|
+      ThinkingSphinx::Processor.new(model: model, id: id).delete
+    end
+  end
```

Comment:
I would need some help to understand this changes and how they relate to cms pages

Reply:
This code was removed from parent class, so it had to move here. No need to keep these methods in parent if they are not used there.

Reply:
No, because here we also remove associated buyers when provider is marked to be removed (but not actually removed to allow their own callbacks to kick in)